packages/react-meteor-data/useTracker.js

```diff
@@ -0,0 +1,188 @@
+function checkCursor(data) {
+  let shouldWarn = false;
+  if (Package.mongo && Package.mongo.Mongo && data && typeof data === 'object') {
+    if (data instanceof Package.mongo.Mongo.Cursor) {
+      shouldWarn = true;
+    } else if (Object.getPrototypeOf(data) === Object.prototype) {
+      Object.keys(data).forEach((key) => {
+        if (data[key] instanceof Package.mongo.Mongo.Cursor) {
+          shouldWarn = true;
+        }
+      });
+    }
+  }
+  if (shouldWarn) {
+    warn(
+      'Warning: your reactive function is returning a Mongo cursor. '
+      + 'This value will not be reactive. You probably want to call '
+      + '`.fetch()` on the cursor before returning it.'
+    );
+  }
+}
+
+// Used to create a forceUpdate from useReducer. Forces update by
+// incrementing a number whenever the dispatch method is invoked.
+const fur = x => x + 1;
+
+function useTracker(reactiveFn, deps, computationHandler) {
+  const { current: refs } = useRef({
+    isMounted: null,
+    doDeferredRender: false
+  });
+
+  const [, forceUpdate] = useReducer(fur, 0);
+
+  const dispose = () => {
+    if (refs.computationCleanup) {
+      refs.computationCleanup();
+      delete refs.computationCleanup;
+    }
+    if (refs.computation) {
+      refs.computation.stop();
+      refs.computation = null;
+    }
+  };
+
+  const runReactiveFn = (c) => {
+    const data = reactiveFn(c);
+    if (Meteor.isDevelopment) checkCursor(data);
+    refs.trackerData = data;
+  };
+
+  const tracked = (c) => {
+    if (c === null || c.firstRun) {
+      // If there is a computationHandler, pass it the computation, and store the
+      // result, which may be a cleanup method.
+      if (computationHandler) {
+        const cleanupHandler = computationHandler(c);
+        if (cleanupHandler) {
+          if (Meteor.isDevelopment && typeof cleanupHandler !== 'function') {
+            warn(
+              'Warning: Computation handler should return a function '
+              + 'to be used for cleanup or return nothing.'
+            );
+          }
+          refs.computationCleanup = cleanupHandler;
+        }
+      }
+      // This will capture data synchronously on first run (and after deps change).
+      // Don't run if refs.isMounted === false. Do run if === null, because that's the first run.
+      if (refs.isMounted === false) {
+        return;
+      }
+      if (refs.isMounted === null) {
+        refs.isMounted = false;
+      }
+      runReactiveFn(c);
+    } else {
+      // If deps are anything other than an array, stop computation and let next render handle reactiveFn.
+      // These null and undefined checks are optimizations to avoid calling Array.isArray in these cases.
+      if (deps === null || deps === undefined || !Array.isArray(deps)) {
+        dispose();
+      } else if (refs.isMounted) {
+        // Only run the reactiveFn if the component is mounted.
+        runReactiveFn(c);
```

Comment:
can `forceUpdate` be moved into this if clause?

Reply:
The logic is a little different there, for whether to call forceUpdate. I actually had some logic to avoid calling `forceUpdate` for unmounted components, but it didn't cause any problems to just call it in every case, and some of the older tests were failing with the logic in place. So I just left it. I can try it again, but it didn't seem important.

Reply:
Actually, it was easier than I thought - that's what I get for coding at 2:00am. This is adjusted as suggested in the latest commit.